data/mods/gen1/moves.js

```diff
@@ -943,12 +943,15 @@ let BattleMovedex = {
 					return;
 				}
 				if (move.volatileStatus && target === source) return;
-				let damage = this.getDamage(source, target, move);
-				if (!damage) return null;
-				damage = this.runEvent('SubDamage', target, source, move, damage);
-				if (!damage) return damage;
-				target.volatiles['substitute'].hp -= damage;
-				source.lastDamage = damage;
+				// NOTE: In future generations the damage is capped to the remaining HP of the
+				// Substitute, here we deliberately use the uncapped damage when tracking lastDamage etc.
+				let uncappedDamage = this.getDamage(source, target, move);
```

Comment:
I could just name this `damage` now that I've inlined `cappedDamage` and theres no need to differentiate, but I want to make it very clear/look different from how it works in later generations.